packages/react/src/__tests__/ReactChildren-test.js

```diff
@@ -868,6 +868,25 @@ describe('ReactChildren', () => {
     ]);
   });
 
+  it('does not warn for flattened positional children', async () => {
+    function ComponentRenderingFlattenedChildren({children}) {
+      return <div>{React.Children.toArray(children)}</div>;
+    }
+
+    const container = document.createElement('div');
+    const root = ReactDOMClient.createRoot(container);
+    await expect(async () => {
+      await act(() => {
+        root.render(
+          <ComponentRenderingFlattenedChildren>
+            <div />
+            <div />
+          </ComponentRenderingFlattenedChildren>,
+        );
+      });
+    }).toErrorDev([]);
+  });
```

Comment:
Maybe add another example where it does still correctly warn if an array was used without keys?

Reply:
we should have test for this already

Reply:
Surprisingly, we don't for the simple case. (We do for iterables of children.)